index.js

```diff
@@ -70,8 +61,14 @@ module.exports = (input, opts) => {
 				task: () => del('node_modules')
 			},
 			{
-				title: `Installing dependencies${opts.yarn ? ' using Yarn' : ''}`,
-				task: install(bin, opts.yarn)
+				title: 'Installing dependencies using Yarn',
+				enabled: () => opts.yarn === true,
+				task: () => exec('yarn', ['install'])
```

Comment:
I mentioned this issue a few times in the PR, i'll quote some relevant sources for context:

>If I have a dependency I cannot install, for instance a OSX only module on the Windows platform, it also triggers a fail-over. execa gives an ENOENT code whether the command could not be found or the installation failed (on Windows at least). It appears to me the only difference in output is the stderr. I could create a regex to differentiate the two, but that seems hacky.

>I thought more about this issue, I'm becoming more convinced we should really just throw the error and let the user deal with it rather than attempting an npm fail-over. If there is a lock file and they don't have Yarn installed (causing a fail-over), this might silently break modules with non-pinned versions, leading to issues down the road. A user should explicitly state if they want to abandon the lock using the --no-yarn flag.

>I went ahead and applied this behavior to my branch. I think the user should explicitly state they want to abandon the lock using --no-yarn rather than silently failing to npm.

Conclusion:
- Difficult to detect the binary is missing versus a standard error with the current library
- If the binary is missing in that scenario, we shouldn't fallback to `npm` because it could silently break the build due to using non-pinned versions
- Users should explicitly state they want to abandon the lock using `--no-yarn` to avoid the aforementioned issues, this allows for the fail-over you described, but not automatically.